src/azul/service/manifest_service.py

```diff
@@ -714,11 +715,75 @@ def write_to(self, output: IO[str]) -> Optional[str]:
             # bundle UUID. Because a file can belong to multiple bundles we use
             # the one with the most recent version.
             bundle = max(doc['bundles'], key=itemgetter('version', 'uuid'))
-            output_name = bundle['uuid'] + '/' + name
+            output_name = self._sanitize_path(bundle['uuid'] + '/' + name)
             output.write(f'url={self._option(url.url)}\n'
                          f'output={self._option(output_name)}\n\n')
         return None
 
+    disallowed_characters = re.compile(r'[\x00-\x1f\\]')
+
+    replaceable_characters = re.compile(r'[<>:"|?*]')
+
+    valid_regex_sequence = re.compile(r'''
+        (\.?[^./ ])([^/]*[^./ ])? # sequences with '/', ' ', '.' zero or more times can compromise system files & paths,
```

Comment:
Line too long. Still awkward. Each of these characters has an English name that you could use. "Sequences with … zero or more times" is convoluted. Consider "consecutive".